R/core_use_gitlab.R

```diff
@@ -0,0 +1,118 @@
+#' }
+use_gitlab <- function(pkg = ".", auth_token = "xxxx", rocker = "verse", rmd_to_knit = "path_to_rmd") {
+  if (auth_token == "xxxx") {
+    stop('This function fails without setting auth_token to your GitLab.com personal access token, e.g., `auth_token = "abcd12345"`. See `?rrtools::use_gitlab` for more details.')
+  }
+
+  if (is.null(git2r::discover_repository(usethis::proj_get()))) {
+    stop("You have not initialized the local git repository yet.")
+  }
+  pkg <- as.package(pkg)
+
+  # gather relevant information for remote git repo
+  if (is.null(getOption("gitlab.username"))) {
+    username <- readline(prompt = "   ...   Please enter your GitLab username: ")
+    options(gitlab.username = username)
+  } else {
+    username <- getOption("gitlab.username")
+  }
+    pkgname <- gsub(pattern = "Package: ",
+                  replacement = "",
+                  readLines("DESCRIPTION")[1])
+
+  # get R version for rocker/r-ver
+  si <- utils::sessionInfo()
+  r_version <- paste0(si$R.version$major, ".", si$R.version$minor)
+
+  # get path to Rmd file to knit
+  if(rmd_to_knit == "path_to_rmd"){
+    dir_list   <- list.dirs()
+    paper_dir  <- dir_list[grep(pattern = "/paper", dir_list)]
+    rmd_path   <- regmatches(paper_dir, regexpr("analysis|vignettes|inst", paper_dir))
+    rmd_parent_path <- file.path(rmd_path, "paper")
+    rmd_path <-  file.path(rmd_path, "paper/paper.Rmd")
+  } else {
+    #  preempt the string with home directory notation or back-slash (thx Matt Harris)
+    rmd_path <- gsub("^.|^/|^./|^~/","",rmd_to_knit)
+  }
+
+  # attempt to push the current branch and set the remote as upstream
+  if (length(grep("^origin\thttps", system("git remote --verbose", intern = TRUE)[2])) > 0) {
+    stop("There is already an `origin` remote associated with this local repo. Please remove it and try again.")
+  } else {
+    system(paste0("git push --set-upstream https://oauth2:", auth_token, "@gitlab.com/", username, "/", pkgname, ".git master"))
+    system(paste0("git remote add origin https://gitlab.com/", username, "/", pkgname, ".git"))
+    git2r::config(branch.master.remote = "origin")
+
+    # assign variables for whisker
+    gh <- github_info(pkg$path)
+    gh$r_version <- r_version
+    gh$rocker <- rocker
+    gh$rmd_path <- rmd_path
+    gh$rmd_parent_path <- rmd_parent_path
+
+    # use the templated .gitlab-ci.yml file for CI/CD
+    use_template("gitlab-ci.yml",
+                 save_as = ".gitlab-ci.yml",
+                 data = gh,
+                 ignore = TRUE,
+                 open = TRUE,
+                 pkg = pkg,
+                 out_path = "")
+    # use the templated Dockerfile file for CI/CD
+    use_template("Dockerfile",
+                 save_as = "Dockerfile",
+                 data = gh,
+                 ignore = TRUE,
+                 open = TRUE,
+                 pkg = pkg,
+                 out_path = "")
+    # replace the README.md
+    file.rename("CONDUCT.md", "CONDUCT.md.bak")
```

Comment:
As already stated: Here should be a check for existence. I also think a console message that mentions this step would be nice.